lib/openstax/aws/stack.rb

```diff
@@ -338,6 +338,47 @@ def status
       end
     end
 
+    def status_reason
+      begin
+        aws_stack.stack_status_reason
+      rescue Aws::CloudFormation::Errors::ValidationError => ee
+        case ee.message
+        when /Stack.*does not exist/
+          "DOES_NOT_EXIST"
+        else
+          raise
+        end
+      end
+    end
+
+    def events
+      aws_stack.nil? || aws_stack.empty? ? [] : aws_stack.events
+    end
+
+    def failed_status?(check_status)
+      check_status.include?("_FAILED") || check_status == "ROLLBACK_IN_PROGRESS"
+    end
+
+    def latest_failed_events
+      begin
+        ###search in latest deployment events
+        aws_stack.events.each_with_object([]) do |event, array|
+          current_status = event.data.resource_status
+          current_reason = event.data.resource_status_reason || ""
+
+          if(failed_status?(current_status))
+            array << {"status" => current_status, "status_reason" => current_reason} 
+          end
+
+          if(event.data.resource_status_reason == "User Initiated")
+            return array
+          end
+        end
+      rescue
+        raise "No registered events"
```

Comment:
When does this happen?  Seems a little odd to me.  If there are really no events, I wouldn't expect this to raise, but instead to return an empty array.